### PR TITLE
automate docs site deploy on commit to master

### PR DESCRIPTION
**Pull request by navyakanu**

### CODE REVIEW

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+
+jobs:
+  build:
+
+    runs-on: ubuntu-latest
+    
+    steps:
+    - uses: actions/checkout@v1
+    - name: Run a multi-line script
+      run: |
```
Comment (pmlopes) [inline .github/workflows/main.yml]: If you run `npm install` first the there's no need to run the manual install commands bellow

File: .github/workflows/main.yml [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+        mvn -Pcodegen install
```
Comment (pmlopes) [inline .github/workflows/main.yml]: Both maven commands can be merged to a single I think. Just merge all profiles `-Pcodegen,io.vertx,io.reactiverse` and the goals

File: .github/workflows/main.yml [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+on: [push]
+
+jobs:
+  build:
+
+    runs-on: ubuntu-latest
+    
+    steps:
+    - uses: actions/checkout@v1
+    - name: Run a multi-line script
```
Comment (pmlopes) [inline .github/workflows/main.yml]: Perhaps a more descriptive name? Build website perhaps?

File: .github/workflows/main.yml [hunk 4]
```diff
@@ -9,13 +9,13 @@ jobs:
     
     steps:
     - uses: actions/checkout@v1
-    - name: Run a multi-line script
+    - name: build website
       run: |
-        mvn -Pcodegen install
+        npm install -g
         sudo npm install -g typedoc
         sudo npm install verdaccio
```
Comment (pmlopes) [inline .github/workflows/main.yml]: these can go away as they are already installed by the command above.

File: package.json [hunk 5]
```diff
@@ -7,7 +7,8 @@
     "gh-pages": "^2.1.1"
   },
   "dependencies": {
-    "typedoc" : "0.15.0"
+    "typedoc" : "0.15.0",
+    "verdaccio": "4.3.3"
```
Comment (pmlopes) [inline package.json]: verdacio is not needed. It is used during development for verifying the packages

**Comment by navyakanu** [comment]
Not sure of the mvn commands order, as of now there is error in the workflow
 Could not find artifact io.reactiverse:es4x-codegen:jar:0.9.2-SNAPSHOT in sonatype-nexus-snapshots (https://oss.sonatype.org/content/repositories/snapshots)

There are 2 pom.xml file, one in the root and one in the generator folder, is there any restrictions that all the dependencies need to be installed in the root before executing commands in the generator?

**Comment by pmlopes** [comment]
I think the correct way is, that all is driven by `package.json`. In order to do this add the following scripts:

```json
{
  "scripts": {
    "pretsdocs": "mvn clean install",
    "tsdocs": "cd generator && mvn -Pcodegen,io.vertx,io.reactiverse clean generate-sources exec:exec@typedoc",
...
}
```

So the shell script on the github actions would be just:

```
npm install
npm run tsdocs
npm run docs:build
npm run docs:publish
```

If maven is not present in the ubuntu machine (I can add the maven wrapper to the project) but the ubuntu machine needs to have java 8 or above + node installed.

**Pull request closed without merging**